jax_md/partition.py

```diff
@@ -359,7 +359,7 @@ def cell_list_fn(position: Array,
     for k, v in kwargs.items():
       sorted_kwargs[k] = v[sort_map]
 
-    sorted_cell_id = jnp.mod(lax.iota(jnp.int64, N), cell_capacity)
+    sorted_cell_id = jnp.mod(lax.iota(jnp.int32, N), cell_capacity)
```

Comment:
For consistency would you mind using `i32` everywhere?

Reply:
fixed